src/tools/run-make-support/src/symbols.rs

```diff
@@ -42,3 +42,46 @@ pub fn any_symbol_contains(path: impl AsRef<Path>, substrings: &[&str]) -> bool
         false
     })
 }
+
+/// Get a list of symbols that are in `symbol_names` but not the final binary.
+///
+/// The symbols must also match `pred`.
+///
+/// The symbol names must match exactly.
```

Comment:
Suggestion: apparently dyld expects dylib symbol names to be prefixed with `_`, so we should describe this difference.